src/bytes.rs

```diff
@@ -1488,7 +1485,7 @@ impl Borrow<[u8]> for BytesMut {
 
 impl fmt::Write for BytesMut {
     fn write_str(&mut self, s: &str) -> fmt::Result {
-        BufMut::put(self, s);
+        self.extend_from_slice(s.as_bytes());
```

Comment:
Hm, that it doesn't grow when being written to is a little surprising, no? Most other containers that can be written to do so...